src/peft/tuners/tuners_utils.py

```diff
@@ -781,6 +796,86 @@ def _move_adapter_to_device_of_base_layer(self, adapter_name: str, device: Optio
                 adapter_layer[adapter_name] = adapter_layer[adapter_name].to(device)
 
 
+def find_minimal_target_modules(
+    target_modules: list[str] | set[str], other_module_names: list[str] | set[str]
```

Comment:
I would prefer this (`list[str] | set[str]`) to `Union[List[str], Set[str]]` more. Nice.

Reply:
Yeah same here.